### ISSUE TITLE
Expose `fastq_join` command to concatenate (not merge) PE reads

### ISSUE THREAD

**Issue by colinbrislawn** [qiime2/q2-vsearch#98]
Vsearch supports --fastq_join in which "sequences are not merged as with the fastq_mergepairs command, but simply joined with a gap."

**Addition Description**
It would work like the existing [merge-pairs (q2 plugin docs)](https://docs.qiime2.org/2024.5/plugins/available/vsearch/merge-pairs/).

**Current Behavior**
only merging is supported via overlap

**Proposed Behavior**
Add a new semantic type for formally-paired (now concatenated!) Illumina reads
Add warnings and docs about how this differs from most other methods

**Questions**
1. Who wants to make this new semantic type?
2. What programs take discontiguous amplicons as input?

**Refs for vsearch**
Forums [x-ref](https://forum.qiime2.org/t/concatenate-r1-and-r2-for-reads-that-cant-join/23646/).
803f-1392r amplicons is 590 reads, using PE300 [x-ref](https://forum.qiime2.org/t/handling-non-merging-paired-end-803f-1392r-amplicon-sequences-in-qiime-2-for-asv-generation/31196)

**Refs for DADA2**
https://forum.qiime2.org/t/dada2-option-justconcatenate/21661/2
https://forum.qiime2.org/t/classification-of-non-overlapping-reads-treated-with-justconcatenate/17914/7

**Comment by lizgehret** [comment]
Hey @colinbrislawn, thanks for submitting this! People have definitely been requesting this for a while, but this will be a big undertaking. We're going to discuss this in our engineering meeting next week to see what kind of timeline makes sense for getting this implemented!

**Comment by lizgehret** [comment]
Hey @colinbrislawn circling back on this - what are some of the actions you're hoping to utilize this on downstream? Just looking for a better scope for prioritizing this - this would be easy enough to create, but we won't be able to use it with anything as it currently stands.

**Comment by colinbrislawn** [comment]
>we won't be able to use it with anything

Yeah, we also reached this conclusion last time. Discontiguous amplicons break a lot of assumptions, so we may want to put this on hold until we have the downstream pipeline planned.

(If someone has reads like this, they should comment here what they plan to do with them!)

**Comment by lizgehret** [comment]
Gotcha! Yeah agreed, we'll put this on hold for now and see if we receive any replies (or forum posts) in the coming months. I'm going to move this to our 'approved but unscheduled' project board in the meantime!

**Comment by colinvwood** [comment]
@miiixi1in is going to start working on this since we have `FeatureData[LinkedSequence]` now

**Comment by colinvwood** [comment]
Update: I was confused, `merge_pairs` is outputting `SampleData[...]` not `FeatureData[...]`, so we can't use the new linked sequence type here. However, given that we are working on support for `FeatureData[LinkedSequence]`s we might want to consider adding and supporting something like `SampleData[LinkedSequencesWithQuality]`. We'll discuss this and I'll circle back.
